debug_toolbar/middleware.py

```diff
@@ -123,6 +144,64 @@ def __call__(self, request):
                 response["Content-Length"] = len(response.content)
         return response
 
+    async def __acall__(self, request):
+        # Decide whether the toolbar is active for this request.
+        show_toolbar = get_show_toolbar()
+        if not show_toolbar(request) or DebugToolbar.is_toolbar_request(request):
+            response = await self.get_response(request)
+            return response
+
+        toolbar = DebugToolbar(request, self.get_response)
+
+        # Activate instrumentation ie. monkey-patch.
+        for panel in toolbar.enabled_panels:
+            panel.enable_instrumentation()
+        try:
+            # Run panels like Django middleware.
+            print(request)
+            response = await toolbar.process_request(request)
+        finally:
+            clear_stack_trace_caches()
+            # Deactivate instrumentation ie. monkey-unpatch. This must run
+            # regardless of the response. Keep 'return' clauses below.
+            for panel in reversed(toolbar.enabled_panels):
+                panel.disable_instrumentation()
+
+        # Generate the stats for all requests when the toolbar is being shown,
+        # but not necessarily inserted.
+        for panel in reversed(toolbar.enabled_panels):
+            panel.generate_stats(request, response)
+            panel.generate_server_timing(request, response)
+
+        # Always render the toolbar for the history panel, even if it is not
+        # included in the response.
+        rendered = toolbar.render_toolbar()
+
+        for header, value in self.get_headers(request, toolbar.enabled_panels).items():
+            response.headers[header] = value
+
+        # Check for responses where the toolbar can't be inserted.
+        content_encoding = response.get("Content-Encoding", "")
+        content_type = response.get("Content-Type", "").split(";")[0]
+        if (
+            getattr(response, "streaming", False)
+            or content_encoding != ""
+            or content_type not in _HTML_TYPES
+        ):
+            return response
+
+        # Insert the toolbar in the response.
+        content = response.content.decode(response.charset)
+        insert_before = dt_settings.get_config()["INSERT_BEFORE"]
+        pattern = re.escape(insert_before)
+        bits = re.split(pattern, content, flags=re.IGNORECASE)
+        if len(bits) > 1:
+            bits[-2] += rendered
+            response.content = insert_before.join(bits)
+            if "Content-Length" in response:
+                response["Content-Length"] = len(response.content)
+        return response
+
```

Comment:
I think all of that is copy-pasted from `def __call__`? If yes, move all of it to a postprocess method, for example `def _postprocess(self, request, response)`, and call this from both `__call__` and `__acall__`.